src/json-patch-duplex.ts

```diff
@@ -524,6 +524,58 @@ module jsonpatch {
     return true;
   }
 
+  
+  /**
+   * Apply a single json-patch on an object tree
+   * Returns the result object.
+   */
+  export function applyPatch(tree: any, patch: any, validate = false): any {
```

Comment:
Can we default `validate` to `false` if `typeof validate !== 'boolean'`? Otherwise the function doesn't work as a reducer because the index is passed as a third parameter and would be coerced to boolean.